### PR TITLE
jibri: init at 8.0-93-g51fe7a2

### PR DESCRIPTION
**Pull request by illustris**
###### Motivation for this change
https://github.com/NixOS/nixpkgs/issues/132623

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/jibri/default.nix [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+let
+  pname = "jibri";
+  version = "8.0-93-g51fe7a2";
+  src = fetchurl {
+    url = "https://download.jitsi.org/stable/${pname}_${version}-1_all.deb";
+    sha256 = "1w78aa3rfdc4frb68ymykrbazxqrcv8mcdayqmcb72q1aa854c7j";
+  };
+in
+stdenv.mkDerivation {
+  inherit pname version src;
```
Comment (SuperSandro2000) [inline pkgs/servers/jibri/default.nix:12]:
```suggestion
stdenv.mkDerivation rec {
  pname = "jibri";
  version = "8.0-93-g51fe7a2";

  src = fetchurl {
    url = "https://download.jitsi.org/stable/jibri_${version}-1_all.deb";
    sha256 = "1w78aa3rfdc4frb68ymykrbazxqrcv8mcdayqmcb72q1aa854c7j";
  };
```

File: pkgs/servers/jibri/default.nix [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+  dontBuild = true;
+
+  unpackCmd = "${dpkg}/bin/dpkg-deb -x $src debcontents";
```
Comment (SuperSandro2000) [inline pkgs/servers/jibri/default.nix:16]:
```suggestion
  unpackCmd = "dpkg-deb -x $src debcontents";
```
and add dpkg to nativeBuildInputs.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/servers/jibri/default.nix (hunk 1 above)
Comment (illustris) [reply pkgs/servers/jibri/default.nix]: thanks, I keep forgetting rec

File: pkgs/servers/jibri/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/{bin,opt/jitsi/jibri,etc/jitsi/jibri}
+    mv etc/jitsi/jibri/* $out/etc/jitsi/jibri/
+    mv opt/jitsi/jibri/* $out/opt/jitsi/jibri/
+
+    # make a tiny wrapper script to launch jibri
+    printf "#!/bin/bash\nexec ${jre_headless}/bin/java -jar $out/opt/jitsi/jibri/jibri.jar \"\$@\"\n" > $out/bin/jibri
```
Comment (SuperSandro2000) [inline pkgs/servers/jibri/default.nix:23]:
```suggestion
    echo "#!/bin/bash\nexec ${jre_headless}/bin/java -jar $out/opt/jitsi/jibri/jibri.jar \"\$@\"" > $out/bin/jibri
```
Also I think using makeWrapper is easier here. Let me know if you need help with that.

Comment (illustris) [reply pkgs/servers/jibri/default.nix:23]: Hm... I'm not too familiar with makeWrapper. I'll check it out.

Comment (illustris) [reply pkgs/servers/jibri/default.nix:23]: @SuperSandro2000 also, I don't think replacing printf with echo will work. Echo treats \n as a string literal, printf changes that to a newline.

**Pull request merged**